src/structures/MessageButtonInteraction.js

```diff
@@ -0,0 +1,164 @@
+'use strict';
+
+const APIMessage = require('./APIMessage');
+const Interaction = require('./Interaction');
+const WebhookClient = require('../client/WebhookClient');
+const { Error } = require('../errors');
+const { InteractionResponseTypes } = require('../util/Constants');
+
+/**
+ * Represents a message button interaction.
+ * @extends {Interaction}
+ */
+class MessageButtonInteraction extends Interaction {
+  // eslint-disable-next-line no-useless-constructor
+  constructor(client, data) {
+    super(client, data);
+
+    /**
+     * The message to which the button was attached
+     * @type {?Message|Object}
+     */
+    this.message = data.message ? this.channel?.messages.add(data.message) ?? data.message : null;
+
+    /**
+     * The custom ID of the button which was clicked
+     * @type {string}
+     */
+    this.customID = data.data.custom_id;
+
+    /**
+     * Whether the reply to this interaction has been deferred
+     * @type {boolean}
+     */
+    this.deferred = false;
+
+    /**
+     * Whether this interaction has already been replied to
+     * @type {boolean}
+     */
+    this.replied = false;
+
+    /**
+     * An associated webhook client, can be used to create deferred replies
+     * @type {WebhookClient}
+     */
+    this.webhook = new WebhookClient(this.applicationID, this.token, this.client.options);
+  }
+
+  /**
+   * Defers the reply to this interaction.
+   * @returns {Promise<void>}
+   * @example
+   * // Defer the reply to this interaction
+   * interaction.defer()
+   *   .then(console.log)
+   *   .catch(console.error)
+   * @example
+   * // Defer to update the button to a loading state
+   * interaction.defer(true)
+   *   .then(console.log)
+   *   .catch(console.error);
+   */
+  async defer() {
```

Comment:
Not sure if this needs a separate function, but there should be a way to respond with `DEFERRED_CHANNEL_MESSAGE`